### PR TITLE
Return feeDelta from modifyLiquidity

### LINKED ISSUE #533: Consider returning BalanceDelta fees for `.modifyLiquidity`
### Component

General design optimization (improving efficiency, cleanliness, or developer experience)

### Describe the suggested feature and problem it solves.

Anytime a liquidity position is modified (i.e. increase liquidity, decrease liquidity), the fees are included in the balance delta. This is largely fine, however distinguishing fee revenue vs position liquidity from the delta is a bit tricky.

Example:
* Alice and Bob have positions on the same range, via a periphery contract
* Alice decreases her liquidity, forcing the periphery contract to collect Alice's position liquidity _and_ Alice & Bob's fees
* The periphery contract must perform extra calculations to discern what portion of the balance delta is position tokens so that it can return Alice's position liquidity + fees

### Describe the desired implementation.

Update `IPoolManager.modifyLiquidity()` to return `BalanceDelta feeDelta`:
```solidity
    function modifyLiquidity(PoolKey memory key, ModifyLiquidityParams memory params, bytes calldata hookData)
        external
        returns (BalanceDelta positionDelta, BalanceDelta feeDelta);
```

Update `Pool.modifyLiquidity()` to return and discern `BalanceDelta feeDelta`:
~~`result = result + toBalanceDelta(feesOwed0.toInt128(), feesOwed1.toInt128());`~~
```solidity
return (result, toBalanceDelta(feesOwed0.toInt128(), feesOwed1.toInt128()));
```

Note: the fee delta should still be accounted
```solidity
        (positionDelta, feeDelta) = pools[id].modifyPosition(...);
        _accountPoolBalanceDelta(key, positionDelta + feeDelta);
```

### Describe alternatives.

Do nothing and require position managers either:

* Call `poolManager.modifyLiquidity` twice, where the first call peforms fee collection (`liquidityDelta = 0`)
* Or, before calling `poolManager.modifyLiquidity`, use `LiquidityAmounts.getAmountsForLiquidity()` to determine the principal token amounts for a range + liquidity delta

### Additional context.

Will try benchmarking to see if its worth the change 👍

### PR DESCRIPTION
**Pull request by saucepoint**
## Related Issue
Closes #533 

## Description of changes

Add another return delta from `PoolManager.modifyLiquidity` which distinguishes the deltas with respect to liquidity and fees. 

**Fees are still credited towards liquidity modifications**, however the additional return value will help periphery contracts separate "principal" delta from "fee" delta

### CODE REVIEW

File: src/PoolManager.sol [hunk 1]
```diff
@@ -188,7 +188,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
             })
         );
 
-        _accountPoolBalanceDelta(key, delta);
+        _accountPoolBalanceDelta(key, delta + feeDelta);
```
Comment (saucepoint) [inline src/PoolManager.sol:191]: another option is keeping the credit logic inside `Pool.sol` and this PR mostly becomes bubbling up `feeDelta` from Pool library --> PoolManager --> caller

File: src/libraries/Pool.sol [hunk 2]
```diff
@@ -151,10 +151,10 @@ library Pool {
     /// @notice Effect changes to a position in a pool
     /// @dev PoolManager checks that the pool is initialized before calling
     /// @param params the position details and the change to the position's liquidity to effect
-    /// @return result the deltas of the token balances of the pool
+    /// @return delta the deltas of the token balances of the pool and the delta from fees
```
Comment (hensha256) [inline src/libraries/Pool.sol:154]: can you add a second @return line for 
```
@return feeDelta 
```

File: src/interfaces/IPoolManager.sol [hunk 3]
```diff
@@ -157,10 +157,10 @@ interface IPoolManager is IProtocolFees, IERC6909Claims {
     /// @param key The pool to modify liquidity in
     /// @param params The parameters for modifying the liquidity
     /// @param hookData Any data to pass to the callback, via `IUnlockCallback(msg.sender).unlockCallback(data)`
-    /// @return delta The balance delta of the liquidity
+    /// @return delta The balance delta of the liquidity, and the balance delta of the fee
```
Comment (hensha256) [inline src/interfaces/IPoolManager.sol:160]: sorry here too

**Review by hensha256 — DISMISSED** [review]
lgtm 🚀

**Review by hensha256 — APPROVED** [review]

**Pull request merged**